### DISCUSSION TITLE
Use of Bakta on unbinned contigs?

### DISCUSSION THREAD

**Discussion by Rridley7** [oschwengers/bakta#161] [Q&A] [answered]
Thank you for your work on this useful tool! I was wondering whether Bakta could also be used on a full set of unbinned contigs from a metagenome (above a certain length cutoff obviously), rather than only on the binned MAGs?

**Comment by oschwengers** [comment, marked as answer]
Thanks for this question!

TL;DR
Technically it should be OK, but I would *not* recommend to do it.

Long answer
We have never tested Bakta on metagenome assemblies, so the following answer is a very theoretical one w/o any practical experience. From a mere technical point of view, a metagenome is just a large assembly and therefore, given that enough resources are provided, Bakta should run without issues and provide reasonable results. However, by *reasonable* I mean within the very low expectations because Bakta was developed to annotate *single* *bacterial* genomes, and therefore there are a couple of implications:
1. Since it uses `Prodigal` to predict `CDS`using a coding table `11` by default, for bacterial contigs, this should be OK in most cases. But for all non-prokaryotic contigs it will predict genes that we cannot trust. Even worse, without a taxonomic classification, we cannot know for which contigs we'll get decent gene predictions.
2. Even for bacterial contigs, coding table `11` is not always the correct one.
3. Since within Bakta `Prodigal` trains on all provided sequences expecting a single genome, the training process will result in wrong assumptions on the assumed single genome, and in turn, this will negatively affect subsequent gene predictions.

Hence, although it is technically possible to use Bakta on unbinned and unfiltered metagenomes, I would heavily argument against it.

However, if this is an issue for more users, and the metagenomes would at least be filtered for prokaryotic contigs, we could think about introducing a `--meta` flag which would be passed on to `Prodigal` so it would run in its `meta` mode.

**Reply by Rridley7** [reply]
Got it, this makes sense. I would assume it is best then in that case to predict proteins prior to giving them to Bakta (via Prodigal --meta or otherwise), and simply run the proteins option. Thanks!

**Reply by oschwengers** [reply]
If you're only interested in the proteins, then yes, that is a better option.

**Reply by EdgarTheOrange** [reply]
I can speak only for myself, but I would appreciate very much if there would be a "meta" mode in future versions!

**Comment by oschwengers** [comment]
Dear @Rridley7 @EdgarTheOrange 
FYI: I just released [v1.7.0](https://github.com/oschwengers/bakta/releases/tag/v1.7.0) which introduces a `--meta` mode. You might be interested in giving this a try. It's *not* a full-blown metagenome annotation pipeline, but focusing on bacterial features, only. It adjusts some internal parameters, mainly activating Pyrodigal's metagenome gene prediction mode.

Have fun and of course, feedback of all kinds is highly welcome and appreciated;-)

**Reply by EdgarTheOrange** [reply]
Thank you! it's very appreciated

**Reply by Rridley7** [reply]
I'll check this out, thank you! One question I had on this, you mentioned in #48  that Bakta is unable to handle archaeal genomes, I believe prodigal's meta (and possibly single also) modes can handle any prokaryotic content. 
What parameters within Bakta would be not suitable for archaeal annotation if this is the case? Within metagenome data, contigs can definitely also be from archaeal lineages depending on the sample, though most current samples usually are bacteria.

**Reply by oschwengers** [reply]
100% d'accord for the gene prediction part but Bakta executes tRNAscan-SE (tRNA) in bacteria mode and its databases only contains bacterial rRNA/ncRNA models and protein sequences. Of course, a larger proportion of universally conserved proteins might be detected but that would happen more by chance and not on purpose.

In addition there might be some subtle differences in bacterial and archaeal genomes that should maybe be addressed, as for example different gene overlap characteristics, translational exceptions, etc. It might be the case, that these differences aren't that important, but I tend to think that we should address archaea with a dedicated fork of Bakta to have the room and chance to properly handle this in a lean manner.

Also, a *real* metagenome version of Bakta would need significant internal changes and a much more comprehensive database. This would def. be addressed by a dedicated tool.

**Reply by Rridley7** [reply]
I see, thanks for this explanation! That makes sense why a separate fork would be useful for these changes.
